src/utils/hook.ts

```diff
@@ -375,6 +375,30 @@ export function useChartQueryWithCache<T>(
   })
 }
 
+export const useAnimationFrame = (callback: () => void, running: boolean = true) => {
+  const savedCallback = useRef(callback)
+  const requestId = useRef(0)
+
+  useEffect(() => {
+    savedCallback.current = callback
+  })
+
+  useEffect(() => {
+    function tick() {
+      savedCallback.current()
+      if (running) {
+        requestId.current = window.requestAnimationFrame(tick)
+      }
+    }
+    if (running) {
+      const animationFrame = window.requestAnimationFrame || window.webkitRequestAnimationFrame
```

Comment:
Can simplify:
```
useEffect(() => {
  if (!running) return

  let requestId = 0
  function tick() {
    savedCallback.current()
    requestId = window.requestAnimationFrame(tick)
  }
  requestId = window.requestAnimationFrame(tick)

  return () => window.cancelAnimationFrame(requestId)
}, [running])
```

Regarding `window.requestAnimationFrame || window.webkitRequestAnimationFrame`, this compatibility should not be needed now, and if it is, it should be implemented at compile time.